hs-abci-sdk/src/Tendermint/SDK/Modules/Bank/Query.hs

```diff
@@ -0,0 +1,43 @@
+module Tendermint.SDK.Modules.Bank.Query where
+
+import           Control.Lens                       ((^.))
+import qualified Data.ByteArray.Base64String        as Base64
+import           Polysemy
+import           Servant.API
+import qualified Tendermint.SDK.BaseApp             as BaseApp
+import qualified Tendermint.SDK.Modules.Auth        as Auth
+import           Tendermint.SDK.Modules.Bank.Keeper (getCoinBalance)
+import           Tendermint.SDK.Types.Address       (Address)
+
+--------------------------------------------------------------------------------
+-- | Query Api
+--------------------------------------------------------------------------------
+
+type GetAddressCoinBalance =
+     "balance"
+  :> QueryParam' '[Required, Strict] "address" Address
```

Comment:
let's go ahead and use `QA Address` here since someone might be querying for historical balances. We can't actually implement that at the moment, but we can go ahead and use the type